### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by serhii-lekariev**
This PR updates the To-Do list application to use the latest Spine libraries.

#### Angular client updates

It also updates the version of Angular to `10`.

Some of the changes in this PR were performed automatically, running `ng update`, which is the recommended way to update Angular-based applications.

#### Deployment modules updates

Common functionality between servers backed by SQL databases was refactored into a `rdbms-common` module, found in the `deployment` directory.

### CODE REVIEW

**Comment by serhii-lekariev** [comment]
@armiol, PTAL.

File: client/angular/src/app/task-creation-wizard/service/task-creation-wizard.service.ts [hunk 1]
```diff
@@ -315,6 +322,13 @@ export class TaskCreationWizard {
     return cmd;
   }
 
+  // tslint:disable-next-line:member-ordering
```
Comment (armiol) [inline client/angular/src/app/task-creation-wizard/service/task-creation-wizard.service.ts:325]: Please comment on the suppression.

File: client/angular/src/app/task-creation-wizard/service/task-creation-wizard.service.ts [hunk 2]
```diff
@@ -194,16 +197,16 @@ export class TaskCreationWizard {
     cmd.setId(this._id);
     const cancelProcess = new Promise<void>((resolve, reject) =>
         this.spineWebClient.command(cmd)
-                           .onOk(resolve)
-                           .onError(reject)
-                           .onRejection(reject)
-                           .post()
+            .onOk(resolve)
```
Comment (armiol) [inline client/angular/src/app/task-creation-wizard/service/task-creation-wizard.service.ts:200]: Please address the layout changes here and in all similar cases.

File: deployment/appengine-web/src/main/java/io/spine/examples/todolist/server/Application.java [hunk 3]
```diff
@@ -125,8 +125,8 @@ private static FirebaseClient firebaseClient() {
         return client;
     }
 
-    private static
-    FirebaseQueryBridge newQueryBridge(QueryService queryService, FirebaseClient firebaseClient) {
+    private static FirebaseQueryBridge newQueryBridge(QueryService queryService,
```
Comment (armiol) [inline deployment/appengine-web/src/main/java/io/spine/examples/todolist/server/Application.java:128]: Please revert.

File: deployment/local-cloud-sql/src/main/java/io/spine/examples/todolist/server/LocalCloudSqlServer.java [hunk 4]
```diff
@@ -152,14 +154,14 @@ private static DataSource createDataSource(String[] args) {
...
      * <p>The value will be obtained from the {@link #properties}.
      *
-     * <p>If the {@link Environment#isTests() environment} is tests,
-     * the method returns prefix for connection to an in-memory database.
+     * <p>If the environment is {@linkplain Tests tests}, the method returns prefix for connection
+     * to an in-memory database.
      *
      * @return the prefix for a connection {@code URL}
      */
     private static String dbUrlPrefix() {
```
Comment (armiol) [inline deployment/local-cloud-sql/src/main/java/io/spine/examples/todolist/server/LocalCloudSqlServer.java:162]: Please eliminate the code duplication in this and `ComputeCloudSqlServer.java` files.

**Comment by serhii-lekariev** [comment]
@armiol, PTAL again.

**Comment by serhii-lekariev** [comment]
@armiol, PTAL again.

**Comment by serhii-lekariev** [comment]
@armiol, PTAL.

**Comment by serhii-lekariev** [comment]
@armiol, PTAL again.

**Pull request merged**